grammar.js

```diff
@@ -1094,6 +1117,96 @@ module.exports = grammar({
       ),
     )),
 
+    _with_settings: $ => seq(
+          field('name', $.identifier),
+          optional('='),
+          field('value', choice($.identifier, alias($._single_quote_string, $.literal))),
+    ),
+
+    create_database: $ => seq(
+      $.keyword_create,
+      $.keyword_database,
+      $.identifier,
+      optional($.keyword_with),
+      repeat(
+        $._with_settings
+      ),
+    ),
+
+    create_role: $ => seq(
+      $.keyword_create,
+      choice(
+        $.keyword_user,
+        $.keyword_role,
+        $.keyword_group,
+      ),
+      $.identifier,
+      optional($.keyword_with),
+      repeat(
+        choice(
+          $._user_access_role_config,
+          $._role_options,
+        ),
+      ),
+    ),
+
+    _role_options: $ => choice(
+      field("option", $.identifier),
+      seq(
+        $.keyword_valid,
+        $.keyword_until,
+        field("valid_until", alias($._literal_string, $.literal))
+      ),
+      seq(
+        $.keyword_connection,
+        $.keyword_limit,
+        field("connection_limit", alias($._integer, $.literal))
+      ),
+      seq(
+        optional($.keyword_encrypted),
+        $.keyword_password,
+        choice(
+          field("password", alias($._literal_string, $.literal)),
+          $.keyword_null,
+        ),
+      ),
+    ),
+
+    _user_access_role_config: $ => seq(
+      choice(
+        seq(optional($.keyword_in), $.keyword_role),
+        seq($.keyword_in, $.keyword_group),
+        $.keyword_admin,
+        $.keyword_user,
+      ),
+      comma_list($.identifier, true),
+    ),
+
+    create_sequence: $ => seq(
+      $.keyword_create,
+      optional(
+        choice(
+          choice($.keyword_temporary, $.keyword_temp),
+          $.keyword_unlogged,
+        )
+      ),
+      $.keyword_sequence,
+      optional($._if_not_exists),
+      $.object_reference,
+      optional(seq($.keyword_as, $._type)),
+      optional(seq($.keyword_increment, optional($.keyword_by), field("increment", alias($._integer, $.literal)))),
+      optional(seq($.keyword_minvalue, choice($.literal, seq($.keyword_no, $.keyword_minvalue)))),
+      optional(seq($.keyword_maxvalue, choice($.literal, seq($.keyword_no, $.keyword_maxvalue)))),
+      optional(seq(
+        $.keyword_start,
+        optional($.keyword_with),
+        field("start", alias($._integer, $.literal)),
+        optional(seq($.keyword_cache, field("cache", alias($._integer, $.literal)))),
+        optional(seq(optional($.keyword_no), $.keyword_cycle))
+      )),
```

Comment:
This section is shared with `alter sequence``
I cannot break this out because they are all optionals and tree-sitter cannot match on potentially empty nodes. Any ideas?

Reply:
cache and cycle clauses are distinct from start! the `cache` and `cycle` keywords respectively are required

Reply:
oh whoops I misidentified what you were talking about, the clauses are correct in alter but still combined here. As for consolidating with `alter` I'm not sure that's practical unfortunately -- the alter version has a choice with additional options (ownership, renaming) that don't apply

Reply:
I was trying to break out the parts that are shared among create and alter sequence:
```
_subset: $ => seq(
  optional(seq($.keyword_as, $._type)),
  optional(seq($.keyword_increment, optional($.keyword_by), $.literal)),
  optional(seq($.keyword_minvalue, choice($.literal, seq($.keyword_no, $.keyword_minvalue)))),
  optional(seq($.keyword_maxvalue, choice($.literal, seq($.keyword_no, $.keyword_maxvalue)))),
  optional(seq($.keyword_start, optional($.keyword_with), field("start", alias($._integer, $.literal)))),
),
```

but this will not work, because it can match on an empty string. So, we could also keep it as it is. I mean this subset will only be used in by two statements. I could live with that.

Reply:
yeah I think the only way you could separate an anonymous node is if you captured the $.object_reference too, but then alter has other options and you lose the `choice`

you might be able to break it out into a _function_ sequence_options() that returns the all-optional `seq` though!

Reply:
I think I keep it as it is. A function for a section that will only occur twice does not sound good.